elastic/kibana
# [Detection Engine][Rules] - Adds custom highlighted fields option

## Summary

Original [PR](https://github.com/elastic/kibana/pull/152871) and changes by @RubixSolver99 .  Thanks Carson for doing the heavy lifting here! This PR cleans it up to get over the line.

Allows a user to define which fields to highlight in areas where we currently use "highlighted fields" feature. This includes:

- Exception auto populate - when adding an exception item from an alert we auto populate values
- Alert details - the alert details flyout includes a "Highlighted fields" section. 
- New alert details - the "Highlighted fields" and "Prevalence" sections

Also ensure:

- the field is populating properly in the rule details/creation/edit pages

New rule field being added is `investigation_fields`. Not adding it to the alert schemas as we will always be querying for latest in our UX, not looking to pull the custom highlighted fields from the alert doc. This means that this field will be available in the alert doc under `kibana.alert.rule.parameters["investigation_fields"]`.

### To test

- Go to rule creation
- Under About > Advanced, find "Custom highlighted fields" option and select one or more fields
- Once alerts are generated:
  - Select to view alert details and ensure the selected custom highlighted fields are showing (if there is alert data to populate with)
  - Select to view alert details and ensure the selected custom highlighted fields are showing under the "Prevelance" tab
  - Select to create an exception from alert and ensure that custom highlighted fields are included and populated with alert data 

### Updated UI screenshots
For the below screenshots, I've created a rule with a custom highlighted field of `host.os.name`
#### Rule creation
<img width="1285" alt="Screenshot 2023-08-13 at 10 20 25 PM" src="https://github.com/elastic/kibana/assets/10927944/9fda3a03-f980-458d-964a-3206f262cfcd">
<img width="1014" alt="Screenshot 2023-08-13 at 10 29 14 PM" src="https://github.com/elastic/kibana/assets/10927944/efefb7d3-cad6-4970-8af3-bd875a3ff40c">

#### Rule details
<img width="799" alt="Screenshot 2023-08-13 at 10 24 59 PM" src="https://github.com/elastic/kibana/assets/10927944/2d174f9e-0151-4ab9-a3e0-97b0f076e3cf">

#### Alerts flyout
<img width="1306" alt="Screenshot 2023-08-13 at 10 29 57 PM" src="https://github.com/elastic/kibana/assets/10927944/78ab5b03-9310-4211-98c7-ed92baeb6951">
<img width="434" alt="Screenshot 2023-08-13 at 10 29 39 PM" src="https://github.com/elastic/kibana/assets/10927944/b5abc7b7-a2dc-4587-85d7-d139109fd48b">

#### Exceptions auto-populate
<img width="1427" alt="Screenshot 2023-08-13 at 10 30 16 PM" src="https://github.com/elastic/kibana/assets/10927944/4074308a-8cd3-476b-911c-9988938b0994">


### Checklist
